[github-actions] Cert_5_3_03_AddressQuery.py fails intermittently

```
======================================================================
FAIL: test (__main__.Cert_5_3_3_AddressQuery)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests/scripts/thread-cert/Cert_5_3_03_AddressQuery.py", line 111, in test
    self.assertEqual(self.nodes[MED1].get_state(), 'child')
AssertionError: 'detached' != 'child'
- detached
+ child
----------------------------------------------------------------------
```

https://github.com/openthread/openthread/runs/6731089615?check_suite_focus=true

--- a/src/core/meshcop/timestamp.hpp
+++ b/src/core/meshcop/timestamp.hpp
@@ -141,6 +141,15 @@ public:
     void AdvanceRandomTicks(void);
 
     /**
+     * This method indicates whether the timestamp indicates an MLE Orphan Announce message.
+     *
+     * @retval TRUE   The timestamp indicates an Orphan Announce message.
+     * @retval FALSE  If the timestamp does not indicate an Orphan Announce message.
+     *
+     */
+    bool IsOrphanTimestamp(void) const { return GetSeconds() == 0 && GetTicks() == 0 && GetAuthoritative(); }
+
+    /**
      * This static method compares two timestamps.
      *
      * Either one or both @p aFirst or @p aSecond can be `nullptr`. A non-null timestamp is considered greater than

--- a/src/core/thread/mle.cpp
+++ b/src/core/thread/mle.cpp
@@ -4113,7 +4113,15 @@ void Mle::HandleAnnounce(RxInfo &aRxInfo)
 
     localTimestamp = Get<MeshCoP::ActiveDatasetManager>().GetTimestamp();
 
-    if (MeshCoP::Timestamp::Compare(&timestamp, localTimestamp) > 0)
+    if (timestamp.IsOrphanTimestamp() || MeshCoP::Timestamp::Compare(&timestamp, localTimestamp) < 0)
+    {
+        SendAnnounce(channel);
+
+#if OPENTHREAD_CONFIG_MLE_SEND_UNICAST_ANNOUNCE_RESPONSE
+        SendAnnounce(channel, aRxInfo.mMessageInfo.GetPeerAddr());
+#endif
+    }
+    else if (MeshCoP::Timestamp::Compare(&timestamp, localTimestamp) > 0)
     {
         // No action is required if device is detached, and current
         // channel and pan-id match the values from the received MLE
@@ -4135,14 +4143,6 @@ void Mle::HandleAnnounce(RxInfo &aRxInfo)
         mAttachCounter = 0;
 
         LogNote("Delay processing Announce - channel %d, panid 0x%02x", channel, panId);
-    }
-    else if (MeshCoP::Timestamp::Compare(&timestamp, localTimestamp) < 0)
-    {
-        SendAnnounce(channel);
-
-#if OPENTHREAD_CONFIG_MLE_SEND_UNICAST_ANNOUNCE_RESPONSE
-        SendAnnounce(channel, aRxInfo.mMessageInfo.GetPeerAddr());
-#endif
     }
     else
     {

